app/src/main/java/pl/droidevs/books/removebook/RemoveBookViewModel.java

```diff
@@ -0,0 +1,49 @@
+package pl.droidevs.books.removebook;
+
+import android.arch.lifecycle.LiveData;
+import android.arch.lifecycle.MutableLiveData;
+import android.arch.lifecycle.ViewModel;
+
+import javax.inject.Inject;
+
+import io.reactivex.CompletableObserver;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.schedulers.Schedulers;
+import pl.droidevs.books.R;
+import pl.droidevs.books.model.Book;
+import pl.droidevs.books.repository.BookRepository;
+
+public class RemoveBookViewModel extends ViewModel{
+
+    private BookRepository bookRepository;
+    private MutableLiveData<Integer> errorMessageResource = new MutableLiveData<>();
+
+    @Inject
+    public RemoveBookViewModel(BookRepository bookRepository) {
+        this.bookRepository = bookRepository;
+    }
+
+    public void removeBook(Book book) {
+        bookRepository.remove(book)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe(new CompletableObserver() {
```

Comment:
Nie używałem RxJavy w wersji 2, ale czy to nie będzie analogiczne:
`.subscribe(() -> {}, throwable -> errorMessageResource.postValue(R.string.remove_book_error))`